docs/changelog.md

```diff
@@ -1,5 +1,19 @@
 # Changelog
 
+### 05-March-2021 - 15:28 CET
+
+- [feature] Stop generating packages for apple-clang 9.1.
+- [feature] Raise error if `ConanInvalidConfiguration` is raised from `build()` method.
```

Comment:
It's currently the only way we have to raise if the version of a dependency is not compliant with the recipe, because validate() can't yet access to this information.

Reply:
🤔 I was not aware of that. It was discussed in this issue https://github.com/conan-io/conan-center-index/issues/4062, @SpaceIm is there an open issue in the conan client repo for this?

Reply:
No, but I'm pretty sure that it was discussed in the PR adding `validate()`, but I don't know what was decided on conan client side after that.

Yes: https://github.com/conan-io/conan/pull/8053#pullrequestreview-535366690

Reply:
I check quickly recipes raising `ConanInvalidConfiguration` in `build()` based on their dependencies versions, here is the list:
`approvaltests.cpp`
`crow`
`gdal`
`jinja2cpp`
`libtorrent`
`paho-mqtt-cpp`
`quill`
`simple-websocket-server`
`spdlog`
`spirv-tools`
`vulkan-loader`

Reply:
Currently, we have a [test in the Conan codebase](https://github.com/conan-io/conan/blob/a0d81b3fa18d57af2eda82203c78c79e09570c7a/conans/test/integration/package_id/test_validate.py#L213) that we could use as a workaround. It is not the interface for Conan 2.0, but if needed we can use it and edit the recipe again after we know which one will be the final interface.

Do not modify the recipes in advance, let's think about it for a while. Not sure if it is better to raise from `build()` or to use this workaround. If it becomes a blocker we can revert that change from the CI, it's not a big problem.

Reply:
We (the team) have talked about it and, even if it is not the interface we would like to have in Conan 2.0, the benefits of using `validate()` are worth even if we have to modify those recipes again in the future.